api/src/components/flag/controller.ts

```diff
@@ -236,17 +254,25 @@ export const resolveFlag = async (
 
         const resolveFlag = await flagService.resolveFlag(event.pathParameters.id);
 
-        const publication = await publicationService.get(flag.publicationId);
+        const publication = flag.publication;
+
+        const latestPublishedVersion =
+            publication.versions.find((version) => version.isCurrent && version.currentStatus === 'LIVE') ||
+            publication.versions.find((version) => version.versionNumber === publication.versions.length - 1);
```

Comment:
Flags against DRAFT versions are not allowed, we only need the first condition before ||